instance/models/openedx_instance.py

```diff
@@ -314,24 +325,36 @@ def get_load_balancer_configuration(self, triggered_by_instance=False):
         The triggered_by_instance flag indicates whether the reconfiguration was initiated by this
         instance, in which case we log additional information.
         """
-        if not self.active_appserver:
+        active_appservers = self.get_active_appservers()
+        if not active_appservers.count():
```

Comment:
The most efficient way to find out whether a queryset is empty is to use `.exists()`.  However, since we need to retrieve all appservers further down anyway, we should force the evaluation of the queryset above:

    active_appservers = list(self.get_active_appservers())

This way, we can make do with a single query.  We can check whether the list is empty using `if not active_appserver`, and neither this nor the itertation further down requires any further database queries.